src/ERPNext/Client.hs

```diff
@@ -18,20 +19,52 @@ module ERPNext.Client
   ) where
 
 import Network.HTTP.Client
-import Data.Text
+import Network.HTTP.Types.Header (hAuthorization, Header)
+import Data.Text hiding (map)
+import Data.Text.Encoding (encodeUtf8)
 import Data.Aeson
 import Data.Proxy qualified
+import ERPNext.Client.Filters (Filters, makeFiltersText)
+import Prelude
 
 -- | Type class for types which represent an ERPNext DocType.
 -- Each DocType has a unique name.
 class IsDocType a where
   docTypeName :: Text
 
-getDocTypeList :: forall a. (IsDocType a, FromJSON a) => Config -> Filters -> ApiResponse [a]
-getDocTypeList = error (unpack $ docTypeName @a)
+getDocTypeList :: forall a. (IsDocType a, FromJSON a) => Config  -> [QueryStringParam]-> IO (ApiResponse [a])
+getDocTypeList config qsParams = do
+  initialRequest <- parseRequest $ unpack $ baseUrl config <> "/resource/" <> docTypeName @a <> "?" <> renderQueryStringParams qsParams
+  let request = initialRequest
```

Comment:
Let's refactor this to a function `createRequest :: Config -> Text -> Text -> IO Request` with args config, path from URL, and HTTP  method (e.g. "GET").